Failures on julia 1.10.0-DEV

```
┌ Debug: running `gcc '-DJULIA_MAIN="MyApp.julia_main"' -DNEW_DEFINE_FAST_TLS_SYNTAX -m64 -march=x86-64 -o /tmp/jl_9OhUSz/MyAppCompiled/bin/MyApp /home/runner/work/PackageCompiler.jl/PackageCompiler.jl/src/embedding_wrapper.c -O2 '-Wl,-rpath,$ORIGIN/../lib:$ORIGIN/../lib/julia' -std=gnu99 -I/opt/hostedtoolcache/julia/nightly/x64/include/julia -fPIC -L/opt/hostedtoolcache/julia/nightly/x64/lib -L/opt/hostedtoolcache/julia/nightly/x64/lib/julia -Wl,--export-dynamic -Wl,-rpath-link,/opt/hostedtoolcache/julia/nightly/x64/lib -Wl,-rpath-link,/opt/hostedtoolcache/julia/nightly/x64/lib/julia -ljulia -ljulia-internal`
└ @ PackageCompiler ~/work/PackageCompiler.jl/PackageCompiler.jl/src/PackageCompiler.jl:190
/home/runner/work/PackageCompiler.jl/PackageCompiler.jl/src/embedding_wrapper.c: In function ‘main’:
/home/runner/work/PackageCompiler.jl/PackageCompiler.jl/src/embedding_wrapper.c:108:24: error: too many arguments to function ‘jl_get_binding_wr’
  108 |     jl_binding_t *bp = jl_get_binding_wr(jl_base_module, jl_symbol("PROGRAM_FILE"), 1);
      |                        ^~~~~~~~~~~~~~~~~
In file included from /home/runner/work/PackageCompiler.jl/PackageCompiler.jl/src/embedding_wrapper.c:7:
/opt/hostedtoolcache/julia/nightly/x64/include/julia/julia.h:1637:28: note: declared here
 1637 | JL_DLLEXPORT jl_binding_t *jl_get_binding_wr(jl_module_t *m JL_PROPAGATES_ROOT, jl_sym_t *var);
      |                            ^~~~~~~~~~~~~~~~~
/home/runner/work/PackageCompiler.jl/PackageCompiler.jl/src/embedding_wrapper.c:109:31: warning: passing argument 2 of ‘jl_checked_assignment’ from incompatible pointer type [-Wincompatible-pointer-types]
  109 |     jl_checked_assignment(bp, firstarg);
      |                               ^~~~~~~~
      |                               |
      |                               jl_value_t * {aka struct _jl_value_t *}
In file included from /home/runner/work/PackageCompiler.jl/PackageCompiler.jl/src/embedding_wrapper.c:7:
/opt/hostedtoolcache/julia/nightly/x64/include/julia/julia.h:1649:71: note: expected ‘jl_module_t *’ {aka ‘struct _jl_module_t *’} but argument is of type ‘jl_value_t *’ {aka ‘struct _jl_value_t *’}
 1649 | JL_DLLEXPORT void jl_checked_assignment(jl_binding_t *b, jl_module_t *mod, jl_sym_t *var, jl_value_t *rhs JL_MAYBE_UNROOTED);
      |                                                          ~~~~~~~~~~~~~^~~
/home/runner/work/PackageCompiler.jl/PackageCompiler.jl/src/embedding_wrapper.c:109:5: error: too few arguments to function ‘jl_checked_assignment’
  109 |     jl_checked_assignment(bp, firstarg);
      |     ^~~~~~~~~~~~~~~~~~~~~
In file included from /home/runner/work/PackageCompiler.jl/PackageCompiler.jl/src/embedding_wrapper.c:7:
/opt/hostedtoolcache/julia/nightly/x64/include/julia/julia.h:1649:19: note: declared here
 1649 | JL_DLLEXPORT void jl_checked_assignment(jl_binding_t *b, jl_module_t *mod, jl_sym_t *var, jl_value_t *rhs JL_MAYBE_UNROOTED);
      |                   ^~~~~~~~~~~~~~~~~~~~~
filter = false: Error During Test at /home/runner/work/PackageCompiler.jl/PackageCompiler.jl/test/runtests.jl:64
  Got exception outside of a @test
  failed process: Process(`gcc '-DJULIA_MAIN="MyApp.julia_main"' -DNEW_DEFINE_FAST_TLS_SYNTAX -m64 -march=x86-64 -o /tmp/jl_9OhUSz/MyAppCompiled/bin/MyApp /home/runner/work/PackageCompiler.jl/PackageCompiler.jl/src/embedding_wrapper.c -O2 '-Wl,-rpath,$ORIGIN/../lib:$ORIGIN/../lib/julia' -std=gnu99 -I/opt/hostedtoolcache/julia/nightly/x64/include/julia -fPIC -L/opt/hostedtoolcache/julia/nightly/x64/lib -L/opt/hostedtoolcache/julia/nightly/x64/lib/julia -Wl,--export-dynamic -Wl,-rpath-link,/opt/hostedtoolcache/julia/nightly/x64/lib -Wl,-rpath-link,/opt/hostedtoolcache/julia/nightly/x64/lib/julia -ljulia -ljulia-internal`, ProcessExited(1)) [1]
  
  Stacktrace:
    [1] pipeline_error
      @ ./process.jl:565 [inlined]
    [2] run(::Cmd; wait::Bool)
      @ Base ./process.jl:480
    [3] run
      @ ./process.jl:477 [inlined]
    [4] run_compiler(cmd::Cmd; cplusplus::Bool)
      @ PackageCompiler ~/work/PackageCompiler.jl/PackageCompiler.jl/src/PackageCompiler.jl:191
    [5] run_compiler
      @ ~/work/PackageCompiler.jl/PackageCompiler.jl/src/PackageCompiler.jl:187 [inlined]
    [6] create_executable_from_sysimg(exe_path::String, c_driver_program::String, julia_main::String)
      @ PackageCompiler ~/work/PackageCompiler.jl/PackageCompiler.jl/src/PackageCompiler.jl:804
    [7] create_app(package_dir::String, app_dir::String; executables::Vector{Pair{String, String}}, precompile_execution_file::String, precompile_statements_file::Vector{String}, incremental::Bool, filter_stdlibs::Bool, force::Bool, c_driver_program::String, cpu_target::String, include_lazy_artifacts::Bool, sysimage_build_args::Cmd, include_transitive_dependencies::Bool)
```

--- a/docs/src/devdocs/binaries_part_2.md
+++ b/docs/src/devdocs/binaries_part_2.md
@@ -175,11 +175,16 @@ int main(int argc, char *argv[])
     jl_set_ARGS(argc, argv);
 
     // Set PROGRAM_FILE to argv[0].
-#if JULIA_VERSION_MAJOR == 1 && JULIA_VERSION_MINOR >= 9
-    jl_binding_t *bp = jl_get_binding_wr(jl_base_module, jl_symbol("PROGRAM_FILE"), 1);
-    jl_checked_assignment(bp, (jl_value_t*)jl_cstr_to_string(argv[0]));
+    jl_sym_t *var = jl_symbol("PROGRAM_FILE");
+    jl_value_t *val = jl_cstr_to_string(argv[0]);
+#if JULIA_VERSION_MAJOR == 1 && JULIA_VERSION_MINOR >= 10
+    jl_binding_t *bp = jl_get_binding_wr(jl_base_module, var);
+    jl_checked_assignment(bp, jl_base_module, var, val);
+#elif JULIA_VERSION_MAJOR == 1 && JULIA_VERSION_MINOR >= 9
+    jl_binding_t *bp = jl_get_binding_wr(jl_base_module, var, 1);
+    jl_checked_assignment(bp, val);
 #else
-    jl_set_global(jl_base_module, jl_symbol("PROGRAM_FILE"), (jl_value_t*)jl_cstr_to_string(argv[0]));    
+    jl_set_global(jl_base_module, var, val);
 #endif
 
     // Set Base.ARGS to `String[ unsafe_string(argv[i]) for i = 1:argc ]`

--- a/src/embedding_wrapper.c
+++ b/src/embedding_wrapper.c
@@ -104,11 +104,15 @@ int main(int argc, char *argv[]) {
     checked_eval_string("append!(empty!(Base.ARGS), Core.ARGS)");
     jl_value_t *firstarg = checked_eval_string("popfirst!(ARGS)");
     JL_GC_PUSH1(&firstarg);
-#if JULIA_VERSION_MAJOR == 1 && JULIA_VERSION_MINOR >= 9
-    jl_binding_t *bp = jl_get_binding_wr(jl_base_module, jl_symbol("PROGRAM_FILE"), 1);
+    jl_sym_t *var = jl_symbol("PROGRAM_FILE");
+#if JULIA_VERSION_MAJOR == 1 && JULIA_VERSION_MINOR >= 10
+    jl_binding_t *bp = jl_get_binding_wr(jl_base_module, var);
+    jl_checked_assignment(bp, jl_base_module, var, firstarg);
+#elif JULIA_VERSION_MAJOR == 1 && JULIA_VERSION_MINOR >= 9
+    jl_binding_t *bp = jl_get_binding_wr(jl_base_module, var, 1);
     jl_checked_assignment(bp, firstarg);
 #else
-    jl_set_global(jl_base_module, jl_symbol("PROGRAM_FILE"), firstarg);    
+    jl_set_global(jl_base_module, jl_symbol("PROGRAM_FILE"), firstarg);
 #endif
     JL_GC_POP();
 

